### PR TITLE
Tuples

### PR DESCRIPTION
**Pull request by traceyyoshima**
Changes:

- Add support to parse tuples.
- Added `JS$Tuple`.
- Replaced `J.Annotations` with `J.Modifier` and `LanguageExtension`.
    - Separated leading and trailing annotations in mapModifiers.
    - Updated applicable methods to handle trailing annotations.
- Cleaned up warnings.

### CODE REVIEW

File: src/main/java/org/openrewrite/javascript/tree/JS.java [hunk 1]
```diff
@@ -1448,6 +1445,81 @@ public TemplateExpression withTag(@Nullable JRightPadded<Expression> tag) {
...
+        @With
+        UUID id;
+
+        @With
+        Space prefix;
+
+        @With
+        Markers markers;
+
+        JContainer<J> types;
```
Comment (knutwannheden) [inline src/main/java/org/openrewrite/javascript/tree/JS.java:1469]: I think we should rename this to `elements` as here: https://github.com/microsoft/TypeScript/blob/d4fbc9b57d9aa7d02faac9b1e9bb7b37c687f6e9/src/compiler/types.ts#L2246-L2249

File: src/test/java/org/openrewrite/javascript/tree/TupleTest.java [hunk 2]
```diff
@@ -59,4 +56,44 @@ void typed() {
...
+                  const [_, ...result] = arg;
+                  return result;
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void trailingCommas() {
```
Comment (knutwannheden) [inline src/test/java/org/openrewrite/javascript/tree/TupleTest.java:90]: Let's also add a test for [named tuple members](https://www.typescriptlang.org/play?ts=4.0.2#example/named-tuples). Were you planning on addressing those separately? If so, let's create an issue for that.

**Review by knutwannheden — COMMENTED** [review]
Looks very good already!

I noticed that we don't have any test for the `readonly` "modifier" (it looks like we might be able to access that using `type.getIntProperty("operator")`). But let's handle that separately, as it will also affect other types.

File: src/test/java/org/openrewrite/javascript/tree/TupleTest.java (hunk 2 above)
Comment (traceyyoshima) [reply src/test/java/org/openrewrite/javascript/tree/TupleTest.java:91]: Ah, thanks for pointing that out!

I used the TS documentation for [reference](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) and thought it was covered by [typed](https://github.com/openrewrite/rewrite-javascript/blob/d045ad8192c0f37632eb062177239efb078aa83d/src/test/java/org/openrewrite/javascript/tree/TupleTest.java#L50), but it wasn't -- updated the test!

**Comment by traceyyoshima** [comment]
> Looks very good already!
> 
> I noticed that we don't have any test for the `readonly` "modifier" (it looks like we might be able to access that using `type.getIntProperty("operator")`). But let's handle that separately, as it will also affect other types.

https://github.com/openrewrite/rewrite-javascript/issues/80

**Pull request merged**